### ISSUE TITLE
Improvement: parse time in event title prompt

### ISSUE THREAD

**Issue by mcorteel** [owncloud/calendar#485]
Hello!

Great work on the new calendar, it's awesome. One time-saving feature I really like in Google calendar is the ability to set the time for an event on creation by typing it before the event title.
For example "14:00 Dentist" will be an event that starts at 14:00, with a default duration (for example 1 hour) with the title "Dentist".

What do you think?

**Comment by georgehrke** [comment]
Hm, the same feature is also in OS X calendar and Fantastical, but I never use it there. In Fantastical it actually annoys me a bit, because it's in the title. (whereas OS X calendar has a dedicated button / input for it).

This is also very difficult, because we would have to adapt different locals. Users will expect it to work with their locale time format.

Also selecting a start time is three clicks at most, so it's not that time-saving.

If anything this would be assigned a very very low priority for now, but I'll let @jancborchardt decide if we want this feature.

@jancborchardt ^

**Comment by jancborchardt** [comment]
Yeah, I would also say it’s low-prio since it’s _a lot_ of work. And when we implement it, people will expect to work it in all kinds of different scenarios and writing styles, otherwise it’s not proper. And since it’s such a fringe feature, there’s a ton of other things which are more important.

@mcorteel contributions are always welcome though of course! :) So if you want to get your feet wet in Calendar development, maybe this is an interesting thing? I suggest to check out some starter issues first though: https://github.com/owncloud/calendar/issues?q=is%3Aissue+is%3Aopen+label%3A%22starter+issue%22

**Comment by mcorteel** [comment]
I am interested and I have considered helping with the development of such a feature. Maybe someday!
I actually met some of you in Berlin two years ago, but I didn't have the time or knowledge to help much. I'm getting there though :)
@georgehrke the number of clicks is exactly my problem. Moving from keyboard to mouse is tiring for a lazy guy like me... But I know that most users don't care!

**Comment by jancborchardt** [comment]
@mcorteel if you are still in Berlin, check out the monthly Berlin meetup: http://meetup.com/owncloud-berlin
Or the upcoming contributor conference! https://owncloud.org/conf

**Comment by mcorteel** [comment]
It would have been great, but I live in Ludwigshafen now!

**Comment by yeKcim** [comment]
"14:00 Dentist" with default hour setting is great
"14:00-16:00 Dentist" is really great

This feature is why I continue to use google calendar :'-(

**Comment by jancborchardt** [comment]
@yeKcim do you have any developer friends? Anyone can contribute here, so if they have time, it would be cool for them to get involved! :)

**Comment by MathieuSchopfer** [comment]
Maybe I can take care of this issue. At least, I have a good idea how to proceed:
1. Add a function to the editor controller, which:
   1. Parse `properties.summary.value` (RegExp mostly)
   2. If necessary, toggle `properties.allDay`
   3. If necessary, update `properties.dtstart.value` and `properties.dtend.value`
2. Trigger that function with a `ng-blur` directive added to the summary input of both editor templates

**Comment by georgehrke** [comment]
I'm very skeptical about this feature.
People will expect it to work with all different locals and we honestly can't put regular expressions on transifex.

**Comment by MathieuSchopfer** [comment]
No, of course it doesn't make much sense to translate regexp ... It means we would have to test for a list of various expressions. Do you fear it would be inefficient ?

**Comment by MathieuSchopfer** [comment]
Regarding the previous comment of @georgehrke, I think this feature should be implemented respecting [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) on date and time formatting. Support for the am/pm designators should also be provided as they are widely used.
This [fiddle](https://jsfiddle.net/MathieuSchopfer/0co07f8p/) is a working example of a function that parses the event title and extract the time schedule. I included recognition of `-` as the interval designator (while the ISO norm recommends `/`). The conversion from string to time is delegated to Moment.js.
